presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -267,6 +267,21 @@ public void testTimestamp()
         // TODO timestamp is not correctly read (see comment in StandardReadMappings.timestampReadMapping), but testing this is hard because of #7122
     }
 
+    @Test
+    public void testJson()
+    {
+        JdbcSqlExecutor jdbcSqlExecutor = new JdbcSqlExecutor(postgreSqlServer.getJdbcUrl());
```

Comment:
Why don't you add roundtrip tests as they are for other data types?
Be sure to include multi-keyd json objects and nested objects/arrays, to cover decoding.

Also, we need tests for `json` created from Presto and for `json`, `jsonb` created from Postgres.

Reply:
I tried to add roundtrip tests, but they failed since json is not supported in ```com.facebook.presto.plugin.jdbcJdbcPageSink``` (Hive doesn't have json type 
and with Postgres I didn't find any ```statement.setXXX``` that is working to insert a json value)
Maybe it's worth to add a new ```com.facebook.presto.tests.datatype.DataSetup``` that returns an inline table (sub query result) in ```com.facebook.presto.tests.sql.TestTable#getName```

Reply:
You're reading json with `ResultSet#getString`, right? Then `Statement#setString` should be useable for writing.

Reply:
Finally succeed to write postgres json value with ```Statement#setString``` after adding ```stringtype=unspecified``` parameter to the JDBC url (https://stackoverflow.com/questions/35844138/how-can-i-insert-json-object-into-postgres-using-java-preparedstatement).

Now tests fails at ```com.facebook.presto.tests.datatype.DataTypeTest#execute``` in ```assertEquals``` since ```materializedRows.getTypes()``` elements are of ```class com.facebook.presto.type.JsonType```  and ```expectedTypes```  elements are of ```class com.facebook.presto.type.spi.JsonType``` ...

Reply:
fixed by extending com.facebook.presto.type.JsonType from com.facebook.presto.spi.type.JsonType and setting it's static JSON attribute to be com.facebook.presto.spi.type.JsonType.JSON